### PR TITLE
Fix `hist` on dataset with binned data

### LINKED ISSUE #2732: Using `hist` on a Dataset containing event data is broken
```Py
import scipp as sc

a = sc.data.binned_x(nevent=1000, nbin=20)
b = sc.data.binned_x(nevent=1500, nbin=20)
ds = sc.Dataset({'a': a, 'b': b})

ds.hist(x=50)
```
yields
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Input In [16], in <cell line: 1>()
----> 1 ds.hist(x=53)

File ~/code/scipp/jupyter/scipp/core/binning.py:270, in hist(x, arg_dict, **kwargs)
    175 def hist(x: Union[_cpp.Variable, _cpp.DataArray, _cpp.Dataset],
    176          arg_dict: Optional[Dict[str, Union[int, Variable]]] = None,
    177          /,
    178          **kwargs: Union[int, Variable]) -> Union[_cpp.DataArray, _cpp.Dataset]:
    179     """Compute a histogram.
    180 
    181     Bin edges can be specified in three ways:
   (...)
    268       {'x': 10, 'y': 5}
    269     """  # noqa #501
--> 270     edges = _make_edges(x, arg_dict, kwargs)
    271     erase = _find_replaced_dims(x, edges)
    272     if isinstance(x, Variable) and len(edges) != 1:

File ~/code/scipp/jupyter/scipp/core/binning.py:161, in _make_edges(x, arg_dict, kwargs)
    159 if arg_dict is not None:
    160     kwargs = dict(**arg_dict, **kwargs)
--> 161 return {name: _parse_coords_arg(x, name, arg) for name, arg in kwargs.items()}

File ~/code/scipp/jupyter/scipp/core/binning.py:161, in <dictcomp>(.0)
    159 if arg_dict is not None:
    160     kwargs = dict(**arg_dict, **kwargs)
--> 161 return {name: _parse_coords_arg(x, name, arg) for name, arg in kwargs.items()}

File ~/code/scipp/jupyter/scipp/core/binning.py:139, in _parse_coords_arg(x, name, arg)
    137 if isinstance(arg, Variable) and name in arg.dims:
    138     return arg
--> 139 coord = _get_coord(x, name)
    140 if isinstance(arg, Integral):
    141     start = coord.min()

File ~/code/scipp/jupyter/scipp/core/binning.py:120, in _get_coord(x, name)
    118 if isinstance(x, Variable):
    119     return x
--> 120 event_coord = x.bins.meta.get(name) if x.bins is not None else None
    121 coord = x.meta.get(name, event_coord)
    122 _require_coord(name, coord)

File ~/code/scipp/jupyter/scipp/core/bins.py:142, in Bins.meta(self)
    139 @property
    140 def meta(self) -> MetaDataMap:
    141     """Coords and attrs of the bins"""
--> 142     return _cpp._bins_view(self._data()).meta

TypeError: _bins_view(): incompatible function arguments. The following argument types are supported:
    1. (arg0: scipp._scipp.core.Variable) -> scipp._scipp.core._BinsViewDataArray

Invoked with: <scipp.Dataset>
Dimensions: Sizes[x:20, ]
Coordinates:
  x                         float64              [m]  (x [bin-edge])  [4.61286e-05, 0.0500006, ..., 0.949181, 0.999135]
Data:
  a                       DataArrayView        <no unit>  (x)  [<scipp.DataArray>
Dimensions: Sizes[row:56, ]
Coordinates:
  x                         float64              [m]  (row)  [0.00313229, 0.0248332, ..., 0.0357545, 0.0465388]
  y                         float64              [m]  (row)  [0.768677, 0.418405, ..., 0.788083, 0.523816]
  z                         float64              [m]  (row)  [0.0579607, 0.789019, ..., 0.398671, 0.942601]
Data:
...
```

Note that using dense data is ok:
```Py
c = sc.data.table_xyz(1000)
d = sc.data.table_xyz(1000)
dense = sc.Dataset({'c': c, 'd': d})
dense.hist(x=50)  # OK
```

### PR DESCRIPTION
**Pull request by nvaytet**
We loop over all elements of the Dataset to determine the bin edges.

Fixes #2732

### CODE REVIEW

File: docs/about/release-notes.rst [hunk 1]
```diff
@@ -59,6 +59,7 @@ Bugfixes
 * Fix a bug in the argument detection when using setitem in combination with value-based slicing `#2750 <https://github.com/scipp/scipp/pull/2750>`_.
 * Fixed broken profile plot on 3-D data `#2746 <https://github.com/scipp/scipp/pull/2746>`_.
 * ``transform_coords`` now works with datasets with only coordinates but no data `#2755 <https://github.com/scipp/scipp/pull/2755>`_.
+* Fixed a bug when using `hist` on a Dataset that contains binned data `#2764 <https://github.com/scipp/scipp/pull/2764>`_.
```
Comment (SimonHeybrock) [inline docs/about/release-notes.rst:62]: Very unspecific. How about ":func:`scipphist` now works with datasets that contain binned data"?

File: tests/binning_test.py [hunk 2]
```diff
@@ -239,6 +239,18 @@ def test_hist_on_dense_data_without_edges_raisesTypeError():
         table.nanhist()
 
 
+def test_hist_on_dataset_with_binned_data():
+    da = sc.data.table_xyz(100)
+    db = sc.data.table_xyz(150)
+    db.coords['y'].values[0] = -5.0
+    edges = sc.linspace(dim='x', start=0.5, stop=1.0, num=21, unit='m')
+    a = da.bin(x=edges)
+    b = db.bin(x=edges)
```
Comment (SimonHeybrock) [inline tests/binning_test.py:248]: `bin` not working on `Dataset`?

Comment (nvaytet) [reply tests/binning_test.py:248]: Not sure I can put them in a dataset before binning, because the coords won't agree?

Comment (SimonHeybrock) [reply tests/binning_test.py:248]: Good point.

**Pull request merged**